sdk/core/azure-core/CMakeLists.txt

```diff
@@ -118,8 +118,11 @@ create_code_coverage(core azure-core azure-core-test)
 # ${CURL_INCLUDE_DIRS} needs to be public as long as we #include<curl.h> in public headers.
 target_include_directories(azure-core PUBLIC ${CURL_INCLUDE_DIRS})
 target_include_directories(azure-core INTERFACE ${nlohmann_json_INCLUDE_DIRS})
-target_link_libraries(azure-core INTERFACE CURL::libcurl Threads::Threads nlohmann_json::nlohmann_json)
+target_link_libraries(azure-core INTERFACE Threads::Threads nlohmann_json::nlohmann_json)
 
+if(BUILD_TRANSPORT_CURL)
```

Comment:
Can you add this same condition in the top at:
https://github.com/Azure/azure-sdk-for-cpp/pull/1218/files#diff-c73316ca9adadb2b459c6af9255dfff8f388c81c73645ea73dcc8df61684dea8R27

Even after removing the linking when libcurl adapter is not built, we are still requiring libcurl to be available and will fail to generate the step when is not.
This will help for Windows users that don't want to use libcurl at all.

Reply:
Let's do that separately @vhvb1989

I am guessing you are referring to:
https://github.com/Azure/azure-sdk-for-cpp/blob/6f3636044e4adf75416c155f78dfeac072160a72/sdk/core/azure-core/CMakeLists.txt#L118-L119

Also, I don't know where your link is supposed to point to, exactly. Do we even expose `curl.h` from any public headers that requires us to need `CURL_INCLUDE_DIRS`?

> we are still requiring libcurl to be available and will fail to generate the step when is not.

Why is this not causing CI to fail currently?

Reply:
No, I am referring to:

https://github.com/Azure/azure-sdk-for-cpp/blob/6f3636044e4adf75416c155f78dfeac072160a72/sdk/core/azure-core/CMakeLists.txt#L24-L31

> that requires us to need CURL_INCLUDE_DIRS

We include it for the source code that includes curl.h

> Why is this not causing CI to fail currently?

Because we install libcurl with vcpkg for all the gates

Reply:
sure thing, I'll add the check